src/components/Layout.tsx

```diff
@@ -18,6 +17,9 @@ const ConnectWalletButton = dynamic(
   () => import("@/components/ConnectWallet/ConnectWalletButton"),
   { ssr: false }
 )
+const NetworkSelector = dynamic(() => import("@/components/NetworkSelector"), {
```

Comment:
Pro tip: NextJS and wagmi have a few quirks still. If you're ever seeing hydration errors on a component like this, you can lazy-load it like this to solve the problem.